From: Felipe Sateler <fsateler@gmail.com>
Date: Thu, 15 Apr 2010 14:48:29 -0400
Subject: [PATCH] Enable build with glibc 2.10

Glibc 2.10 changed the signatures of scandir{,64}. Thanks to Joao Pinto
for the patch.
---
 installwatch/installwatch.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/installwatch/installwatch.c b/installwatch/installwatch.c
index 21c91c7..d34850d 100644
--- a/installwatch/installwatch.c
+++ b/installwatch/installwatch.c
@@ -100,7 +100,7 @@ static int (*true_xstat)(int,const char *,struct stat *);
 static int (*true_lxstat)(int,const char *,struct stat *);
 static int (*true_scandir)(	const char *,struct dirent ***,
 				int (*)(const struct dirent *),
-				int (*)(const void *,const void *));
+				int (*)(const struct dirent **,const struct dirent **));
 static int (*true_symlink)(const char *, const char *);
 static int (*true_truncate)(const char *, TRUNCATE_T);
 static int (*true_unlink)(const char *);
@@ -120,7 +120,7 @@ static int (*true_open64)(const char *, int, ...);
 static struct dirent64 *(*true_readdir64)(DIR *dir);
 static int (*true_scandir64)(	const char *,struct dirent64 ***,
 				int (*)(const struct dirent64 *),
-				int (*)(const void *,const void *));
+				int (*)(const struct dirent64 **,const struct dirent64 **));
 static int (*true_xstat64)(int,const char *, struct stat64 *);
 static int (*true_lxstat64)(int,const char *, struct stat64 *);
 static int (*true_truncate64)(const char *, __off64_t);
@@ -3079,7 +3079,7 @@ int rmdir(const char *pathname) {
 
 int scandir(	const char *dir,struct dirent ***namelist,
 		int (*select)(const struct dirent *),
-		int (*compar)(const void *,const void *)	) {
+		int (*compar)(const struct dirent **,const struct dirent **)	) {
 	int result;
 
 	if (!libc_handle)
@@ -3691,7 +3691,7 @@ struct dirent64 *readdir64(DIR *dir) {
 
 int scandir64(	const char *dir,struct dirent64 ***namelist,
 		int (*select)(const struct dirent64 *),
-		int (*compar)(const void *,const void *)	) {
+		int (*compar)(const struct dirent64 **,const struct dirent64 **)	) {
 	int result;
 
 	if (!libc_handle)
-- 
